### PR TITLE
[docs-only] adding documemtation for graph api for adding members to drive

### PR DESCRIPTION
**Pull request by ghost**
## Description

This PR adds documentation for adding members to space using the graph api with curl examples

## Related Issue

- Relate issue:https://github.com/owncloud/ocis/issues/3139

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: docs/apis/http/graph/spaces.md [hunk 1]
```diff
@@ -638,3 +638,129 @@ The space to be deleted was not disabled before.
 ```
 {{< /tab >}}
 {{< /tabs >}}
+
+## Sharing Space
+
+### Share a Spaces (root) `POST /drives/{drive-id}/root/invite`
```
Comment (saw-jan) [inline docs/apis/http/graph/spaces.md:644]:
```suggestion
### Add Member to Space `POST /drives/{drive-id}/root/invite`
```

File: docs/apis/http/graph/spaces.md [hunk 2]
```diff
@@ -638,3 +638,130 @@ The space to be deleted was not disabled before.
...
+        }
+    ]
+}
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+### Add Member to Space (permissions) `POST /drives/{drive-id}/items/{item-id}/invite`
+
+If you are adding a member to space you should use the previous API or you have to make an extra `PROPFIND` request to get the item-id.
```
Comment (ScharfViktor) [inline docs/apis/http/graph/spaces.md:716]: I would say that we use this endpoint to share space items. but we also can share space using this endpoint and itemId=spaceId

> ### Add Member to Space (permissions)

what means `permissions` here? maybe you wanted to say that we can share space using rolesId or array of the permissions?

File: docs/apis/http/graph/spaces.md [hunk 3]
```diff
@@ -638,3 +638,130 @@ The space to be deleted was not disabled before.
...
+            "objectId": "f7fbf8c8-139b-4376-b307-cf0a8c2d0d9c",
+            "@libre.graph.recipient.type": "user"
+        }
+    ]
+    "expirationDateTime":"2024-07-31T23:59:59.999+05:45",
+}'
+```
+{{< hint type=info title="Body value" >}}
+
+This recipients can be a user or a group. You should provide the following body when sharing with a group
```
Comment (phil-davis) [inline docs/apis/http/graph/spaces.md:668]:
```suggestion
The recipients can be a user or a group. You should provide the following body when sharing with a group:
```

File: docs/apis/http/graph/spaces.md (hunk 2 above)
Comment (saw-jan) [reply docs/apis/http/graph/spaces.md:716]: itemid: space root file-id

**Comment by ScharfViktor** [comment]
> Do we even need to add docs at two places? We already have swagger for the graph APIs

I would leave APIs->https->libreGraph in the documentation since there are other APIs in there as well. I would just replace the method descriptions with links from swagger as suggested by Michael

Screenshot 2024-07-29 at 09 13 59

**Comment by micbar** [comment]
@ScharfViktor This is the way to go IMHO.

**Comment by ScharfViktor** [comment]
> so you are saying i should just add the endpoint related to spaces that are mentioned in swagger and mention the swagger link like:


Yeah

File: docs/apis/http/graph/spaces.md [hunk 4]
```diff
@@ -155,6 +155,10 @@ curl -L -X POST 'https://localhost:9200/graph/v1.0/drives/' \
 {{< /tab >}}
 {{< /tabs >}}
 
+### Create a space item `POST /drives/{drive-id}/root/children`
```
Comment (ScharfViktor) [inline docs/apis/http/graph/spaces.md:158]: Create a space item (Enable sync)

**Review by v-scharf — COMMENTED** [review]
https://github.com/owncloud/ocis/pull/9696/files#diff-61d5430e6ab7e3c19d7491f4258e8baa127c9691c3f6f38237c8d7edd79fe77aR94 

that would be great if we change it too

**Comment by ScharfViktor** [comment]
> > https://github.com/owncloud/ocis/pull/9696/files#diff-61d5430e6ab7e3c19d7491f4258e8baa127c9691c3f6f38237c8d7edd79fe77aR94
> > that would be great if we change it too
> 
> should we only change that? or we can change others too

I would change only it https://github.com/owncloud/ocis/pull/9696/files#diff-61d5430e6ab7e3c19d7491f4258e8baa127c9691c3f6f38237c8d7edd79fe77aR131-R191

and leave the rest of it as it is.

**Comment by ScharfViktor** [comment]
> Ok. If we do that we should remove the [filtering/ordering](https://github.com/owncloud/ocis/pull/9696/files#diff-61d5430e6ab7e3c19d7491f4258e8baa127c9691c3f6f38237c8d7edd79fe77aR192-R254) methods too because it would be in https://owncloud.dev/libre-graph-api/#/me.drives/ListMyDrives

yes, we can

**Review by v-scharf — APPROVED** [review]

**Pull request merged**